### ISSUE TITLE
Separate module for confluence site converter to internal representation with ability to compare by content with already produced output.

### ISSUE THREAD

**Issue by qwazer** [bsorrentino/maven-confluence-plugin#166]
HI @bsorrentino 

I'm thinking about #163 and I have the next proposal:

#### Separate module `docs-to-confluence-format-converter` 

Create separate module `docs-to-confluence-format-converter`  with next features:
1) Receive as input site description in site.xml  or YAML format [1]
2) Works with next formats as input: CONFLUENCE_WIKI, STORAGE, MARKDOWN, ASCIIDOC [2]
3) Produce internal representation of site where
    * content stored in CONFLUENCE_WIKI or STORAGE format
    * each confluence object has metadata object (json?) with internal ID, title, parentPage
4)  Module `docs-to-confluence-format-converter` has 2 main API methods:
    * convert
    * compateConvertedAndLoaded
6) The module externalized for further improvements and possibly reused in `gradle-confluence-plugin`

#### Further improvements:
[1]: YAML is more popular in modern IT: good looking, etc
[2]: ASCIIDOC currently not supported, but it'll be nice to add support for it

**Comment by bsorrentino** [comment]
I agree on module separation ... because, as you say, it could be considered a independent module.

Concerning `YAML` 
> It should be implemented by core module 

Concerning `ASCIIDOC` 
> It should be implemented in such new module

I'll consider this update ... very impacting and requires several updates ... so I think that this should be planned for a release 7. for this reason I'd like to do :

1. Release the final `6.0` version
1. Put release `6.0` in support branch
1. move development on `7.0-SNAPSHOT`
